src/package_tree.rs

```diff
@@ -429,3 +429,226 @@ pub fn get_package_name(path: &str) -> String {
     let bsconfig = read_bsconfig(&path);
     bsconfig.name
 }
+
+fn get_unallowed_dependents(
+    packages: &AHashMap<String, Package>,
+    package_name: &String,
+    dependencies: &Vec<String>,
+) -> Option<String> {
+    for deps_package_name in dependencies {
+        if let Some(deps_package) = packages.get(deps_package_name) {
+            let deps_allowed_dependents = deps_package.bsconfig.allowed_dependents.to_owned();
+            if let Some(allowed_dependents) = deps_allowed_dependents {
+                if !allowed_dependents.contains(package_name) {
+                    return Some(deps_package_name.to_string());
+                }
+            }
+        }
+    }
+    return None;
+}
+
+pub fn validate_packages_dependencies(packages: &AHashMap<String, Package>) -> bool {
+    for (package_name, package) in packages {
+        let bs_dependencies = &package.bsconfig.bs_dependencies.to_owned().unwrap_or(vec![]);
+        let pinned_dependencies = &package.bsconfig.pinned_dependencies.to_owned().unwrap_or(vec![]);
+        let dev_dependencies = &package.bsconfig.bs_dev_dependencies.to_owned().unwrap_or(vec![]);
+
+        if let Some(unallowed_dependency) = get_unallowed_dependents(packages, package_name, bs_dependencies)
```

Comment:
I think you can do something like:

```
let has_unallowed_dependencies = vec![
("Allowed Dependencies", bs_dependencies),
("Pinned Dependencies", pinned_dependencies),
("Dev Dependencies", dev_dependencies),
]
.iter()
.map(|(name, dependencies)| {
  if let Some(unallowed_dependency) =
            get_unallowed_dependents(packages, package_name, dependencies)
        {
            println!(
                "\n{} the package {} is not allowed to have {} as a pinned-dependency. Check the {} entry in the bsconfig.json",
                console::style("Error").red(),
                console::style(package_name).bold(),
                console::style(unallowed_dependency).bold(),
                console::style(String::from(name)).bold().dim()
            );
           false
        } else {
        true
       }
})
.any(|x| x);

if has_unallowed_dependencies { return false; }
```

Thinking of it - perhaps it's better not to return early on the first one we find, but rather return all unallowed dependants from `get_unallowed_dependants` ? 

Not sure how likely that scenario is though

Reply:
I'm not sure either 🤔 
Personally, the complete list of unallowed parents may be too much information at once.
But it's also true that to fix this error it would require a `bsconfig.json` change, and a change to the config won't retrigger the build, so you can't wait for the next error on the `watch` output but you'll need to retrigger a new build manually. 

For now, I've applied the change you've suggested above. 👍

Reply:
I'm looking into the other implementation.
Do you think we should group them by package:
``` 
Package package-a has the following unallowed dependencies:
     bs-dependencies: 
         package-b
         [...]
     pinned-dependencies: 
        package-c
        [...]
```
or by dependency type? 🤔

Reply:
I think doing it by package will be the nicest, yes! Nice!